gateway/ln-gateway/proto/gateway_lnrpc.proto

```diff
@@ -172,3 +189,53 @@ message GetRouteHintsResponse {
   // The route hints to the associated lightning node
   repeated RouteHint route_hints = 1;
 }
+
+message ConnectToPeerRequest {
+  // The public key of the node we're connecting to.
+  string pubkey = 1;
+
+  // The host address of the node we're connecting to.
+  string host = 2;
+}
+
+message GetFundingAddressResponse {
+  // An address belonging to the lightning node's on-chain wallet.
+  string address = 1;
+}
+
+message OpenChannelRequest {
+  // The public key of the node we're opening a channel to.
+  string pubkey = 1;
+
+  // The capacity of the channel, in sats.
+  uint64 channel_size_sats = 2;
+
+  // The amount of sats that should be pushed to the
+  // counterparty once the channel is opened.
+  uint64 push_amount_sats = 3;
+}
+
+message ListActiveChannelsResponse {
+  message ChannelInfo {
+    // The pubkey of the lightning node that the channel is open with.
+    string remote_pubkey = 1;
+
+    // The capacity of the channel, in sats.
+    uint64 channel_size_sats = 2;
+
+    // The SCID of the channel.
+    uint64 short_channel_id = 3;
+  }
```

Comment:
Would be useful to have spendable/receivable local/remote balance to make choices on which channel to keep/close should this API ever be used for channel mgmt.

Reply:
Done! I called the fields `outbound_liquidity_sats` and `inbound_liquidity_sats` since LND and CLN have slightly different APIs and this seemed like the most intuitive implementation-agnostic way to convey channel management information. For LND, we take the balance for a particular channel partner and subtract their channel reserve. For CLN, we use the `spendable_msat` and `receivable_msat` fields from the [`listpeerchannels`](https://docs.corelightning.org/reference/lightning-listpeerchannels) RPC. These fields in CLN seem to behave a bit strangely, often substantially underestimating the available liquidity on one or both sides, but I couldn't find a better way to approach this.